### PR TITLE
samAssembler Part 6: Diploid Base Calling from Ancient DNA Reads

### PR DESCRIPTION
**Pull request by rimangan**
# Description

Alternative parameterization of the samAssembler likelihood functions that allow us to control for postmortem damage-associated cytosine deamination in variant calling.
Please refer to section six in the attached documentation for an explanation of the mathematical framework here.
[samAssembly.pdf](https://github.com/vertgenlab/gonomics/files/11305582/samAssembly.pdf)

## Relevant Links

- [Display Text](https://www.vertgenlab.org/)

### Checklist before requesting a review

- [x] I performed a self-review of my code
- [x] If it this a core feature, I have added thorough tests
- [x] The command `go fmt` was used on all files included

### Testing

None

### Screenshots & Media

None

### Edge cases / Breaking Changes / Known Issues

None

### CODE REVIEW

File: cmd/samAssembler/score.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ func samAssemblerScore(scoreType string, inFileList string, outFile string) {
 		baseMatrixByRefBase(inFileList, outFile, false)
 	case "baseMatrixByRefBase":
 		baseMatrixByRefBase(inFileList, outFile, true)
+	default:
+		log.Fatalf("Unrecognized score mode: %s.\n", scoreType)
```
Comment (edotau) [inline cmd/samAssembler/score.go:24]:
```suggestion
		log.Fatalf("Error: Unrecognized score mode: %s.\n", scoreType)
```

**Review by edotau — APPROVED** [review]
One small log statement suggestion, but from a programmatic stand point enough to get 👍

File: sam/ancientDiploidBase.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	}
+	return logspace.Multiply(firstTerm, logspace.Multiply(secondTerm, logspace.Multiply(thirdTerm, fourthTerm)))
+}
+
+// epsilonOverThreeLikelihoodExpression is a helper function of ancientBaseLikelihood that calculates the expression
+// (epsilon/3.0)^count.
+// Checks the cache to see if this expression has already been calculated. Otherwise, updates the cache.
+func epsilonOverThreeLikelihoodExpression(count int, epsilon float64, cache AncientLikelihoodCache) float64 {
+	if count < len(cache.EpsilonOverThree) { // if the coverage is within the cache bounds
+		if cache.EpsilonOverThree[count] != 0 {
```
Comment (craiglowe) [inline sam/ancientDiploidBase.go:80]: Not a problem, but this logic is repeated in a few functions and could be shortened to:
if cache.Something[count] == 0 { // if cache miss 
     cache.Something[count] = math // put it in the cache
}
return cache.Something[count]

**Review by craiglowe — APPROVED** [review]
Great!
